### PR TITLE
Add collectVnetConfig to GeoMasterDataProvider

### PR DESCRIPTION
**Pull request by michimune**
This new API calls to a geomaster API which gets a collection of VNET related configurations for the specified subscription ID.

### CODE REVIEW

File: src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs [hunk 1]
```diff
@@ -206,6 +206,48 @@ public async Task<VnetValidationRespone> VerifyHostingEnvironmentVnet(string sub
...
+        ///                   vnetRg,
+        ///                   vnetName,
+        ///                   subnet);
+        ///
+        /// }
+        /// </code>
+        /// </example>
+        public async Task<VnetConfiguration> CollectVirtualNetworkConfig(string subscriptionId, string vnetResourceGroup, string vnetName, string vnetSubnetName, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var path = string.Format(@"subscriptions/{0}/providers/Microsoft.Web/collectVnetConfiguration", subscriptionId);
```
Comment (dagrooms52) [inline src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs]: Style: `@` is only necessary for multi-line strings

File: src/Diagnostics.ModelsAndUtils/Models/GeoMaster/NetworkResourceProviderEntities.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    {
+        public string ProvisioningState { get; set; }
+        public string NextHopIpAddress { get; set; }
+        public string AddressPrefix { get; set; }
+        public string NextHopType { get; set; }
+    }
+
+    public class NetworkAddressSpace
+    {
+        public List<string> AddressPrefixes { get; set; }
```
Comment (dagrooms52) [inline src/Diagnostics.ModelsAndUtils/Models/GeoMaster/NetworkResourceProviderEntities.cs]: Let's use `IEnumerable` for iterative needs (foreach) or `IList` if this requires `.Add()`

Comment (michimune) [reply src/Diagnostics.ModelsAndUtils/Models/GeoMaster/NetworkResourceProviderEntities.cs]: That's right

**Comment by hforeste** [comment]
Hey Michi, what do you think about resorting to using ASC to get networking details?

**Comment by michimune** [comment]
It's definitely an option and it's already available, but it's sometimes slow (> 30 seconds). We expect this API to be faster.

**Pull request merged**